code/script.js

```diff
@@ -1,53 +1,273 @@
 }
 
 // A function to start the conversation
 const greetUser = () => {
-  // Here we call the function showMessage, that we declared earlier with the argument:
-  // "Hello there, what's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, what's your name?", 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆 and see what happens
+  showMessage(`Hello there my name is Fikabot, what's your name?`, "bot")
 }
 
-// Eventlisteners goes here 👇
-
-// Here we invoke the first function to get the chatbot to ask the first question when
-// the website is loaded. Normally we invoke functions like this: greeting()
-// To add a little delay to it, we can wrap it in a setTimeout (a built in JavaScript function):
-// and pass along two arguments:
-// 1.) the function we want to delay, and 2.) the delay in milliseconds 
-// This means the greeting function will be called one second after the website is loaded.
+//Call for first greeting, with a short delay
 setTimeout(greetUser, 1000)
+
+//Function that handle the name input
+const handleNameInput = (event) => {
+  event.preventDefault()
+  // Store the value in a variable so we can access it after we
+  const name = nameInput.value
+  if (name !== "") {
+    showMessage(`I'm ${name}`, "user")
+    setTimeout(() => firstChoice(name), 1000)
+  } else {
+    showMessage(`I'm ${name}`, "user")
+    showMessage(`Please type your real name.`, "bot")
+    setTimeout(() => greetUser(), 1000)
+  }
+  // clear value from the input
+  nameInput.value = ""
+}
+
+//Call for handleNameInput func. to activate 1st responce
+submitButton.onclick = handleNameInput
+
+let choice
+
+// Add buttons for choice and handle user choice
+const firstChoice = (name) => {
+  showMessage(`So ${name} what would you like to order?`, "bot")
+  form.innerHTML = `
+  <div class="categories">
+  <button class="order-btn" id="coffee" type="submit">Coffee</button>
+  <button class="order-btn" id="bread" type="submit">Bread</button>
+  <button class="order-btn" id="pastry" type="submit">Pastry</button></div>`
+
+  document
+    .getElementById("coffee")
+    .addEventListener("click", (handleFirstChoice) => {
+      handleFirstChoice.preventDefault()
+      choice = "coffee"
+      showMessage("I would like a coffee, please!", "user")
+      setTimeout(() => handleSecondChoice(choice), 1000)
+    })
+  document
+    .getElementById("bread")
+    .addEventListener("click", (handleFirstChoice) => {
+      handleFirstChoice.preventDefault()
+      choice = "bread"
+      showMessage("I would like some bread, please!", "user")
+      setTimeout(() => handleSecondChoice(choice), 1000)
+    })
+  document
+    .getElementById("pastry")
+    .addEventListener("click", (handleFirstChoice) => {
+      handleFirstChoice.preventDefault()
+      choice = "pastry"
+      showMessage("I would like a pastry, please!", "user")
+      setTimeout(() => handleSecondChoice(choice), 1000)
+    })
+}
+let subChoice
```

Comment:
You can keep your global variables together up top, beneath the DOM selectors. This way, it will be easier to keep track of them